sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,364 @@ def to_rotation_matrix(self, v=None):
+
+        The scalar component of a quaternion is the real part of the quaternion.
+        It gives the rotation angle, and the three imaginary components, when
+        normalized, give the direction ofthe rotation axis.
+
+        Returns
+        =======
+
+        expr: real number representing the scalar part of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.scalar()
+        4
+
+        """
+
+        return self.a
+
+    def vector(self):
+        """
+        Returns the vector part of the quaternion.
+
+        Explanation
+        ===========
+
+        The vector part of a quaternion is also called a versor that is the complex
+        part of that quaternion.The vector part of the quaternion when normalized, gives
+        the rotation axis.
+
+        Returns
+        =======
+
+        Quaternion: represnting vector part of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.vector()
+        0 + 1*i + 1*j + 1*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.vector()
+        0 + 8*i + 13*j + 12*k
+
+        See Also
+        ========
+        https://en.wikipedia.org/wiki/Versor
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d)
+
+    def axis(self):
+        """
+        Returns the axis of the quaternion.
+
+        Explanation
+        ===========
+
+        The axis of a quaternion is the normalized versor (vector part) of that quaternion.
+
+        Returns
+        =======
+
+        Quaternion: representing axis of the provided quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.axis()
+        0 + sqrt(3)/3*i + sqrt(3)/3*j + sqrt(3)/3*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.axis()
+        0 + 8*sqrt(377)/377*i + sqrt(377)/29*j + 12*sqrt(377)/377*k
+
+        See Also
+        ========
+        vector
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d).normalize()
+
+    def is_pure(self):
+        """
+        Returns true if the quaternion is pure else returns false.
+
+        Explanation
+        ===========
+
+        A pure quaternion (also a vector quaternion) is a quaternion with scalar
+        part equal to 0.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.is_pure()
+        False
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.is_pure()
+        True
+
+        """
+
+        return self.a == 0 and not (self.a == self.b == self.c == self.d)
+
+    def angle(self):
+        """
+        Returns the angle of the quaternion measured in the real-axis plane.
```

Comment:
I don't understand what this means from the description

Reply:
The "real-axis plane" is correct. It's actually the "complex plane formed by the scalar and vector parts of the quaternion, embedded in 4D", this can be added to the explanation section.